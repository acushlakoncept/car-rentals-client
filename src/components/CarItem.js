import React from 'react';
import { Card, Button } from 'react-bootstrap';
import ride from '../imgs/ride.jpg';

export default function CarItem() {
  return (
    <Card style={{ width: '18rem' }} className="m-2">
      <Card.Img variant="top" src={ride} />
      <Card.Body>
        <Card.Title className="font-weight-bold">Toyota Corolla 2011 Silver</Card.Title>
        <Card.Text>
          <span className="badge badge-secondary">Automatic</span>
          {' '}
          {' '}
          <span className="badge badge-pill badge-success">AC - Yes </span>
          {' '}
          <span className="badge badge-info">2021-01-18</span>
        </Card.Text>
        <Button className="btn-oval w-100">Hire now</Button>
      </Card.Body>
    </Card>
  );
}
