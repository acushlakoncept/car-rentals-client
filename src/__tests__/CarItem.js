import React from 'react';
import renderer from 'react-test-renderer';
import CarItem from '../components/CarItem';

it('renders CarItem correctly', () => {
  const car = {
    id: 1,
    image: 'https://source.unsplash.com/800x600/?car,black',
    name: 'Chevy Malibu 2020 Grey',
    ac: 'yes',
    date_added: '2021-01-22T19:37:44.393Z',
    transmission: 'Automatic',
  };
  const tree = renderer.create(<CarItem car={car} handleFavourite={jest.fn()} />);
  expect(tree).toMatchSnapshot();
});
