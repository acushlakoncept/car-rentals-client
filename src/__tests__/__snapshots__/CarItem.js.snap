// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CarItem correctly 1`] = `
<div
  className="m-2 card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    src="https://source.unsplash.com/800x600/?car,black"
  />
  <div
    className="card-body"
  >
    <div
      className="font-weight-bold card-title h5"
    >
      <svg
        color="red"
        fill="currentColor"
        height={20}
        onClick={[MockFunction]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "red",
          }
        }
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Add to Favourite
        </title>
        <path
          clipRule="evenodd"
          d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
          fillRule="evenodd"
        />
      </svg>
       
      Chevy Malibu 2020 Grey
    </div>
    <p
      className="card-text"
    >
      <span
        className="badge badge-secondary"
      >
        Automatic
      </span>
       
       
      <span
        className="badge badge-pill badge-success"
      >
        AC -
        yes
      </span>
       
      <span
        className="badge badge-info"
      >
        2021-01-22T19:37:44.393Z
      </span>
    </p>
    <a
      className="btn-oval p-2"
      href="/cars/1"
      onClick={[Function]}
    >
      Learn more
    </a>
  </div>
</div>
`;
